sdk/src/client/stronghold/mod.rs

```diff
@@ -484,6 +494,16 @@ impl StrongholdAdapter {
     ///
     /// [`unload_stronghold_snapshot()`]: Self::unload_stronghold_snapshot()
     pub async fn write_stronghold_snapshot(&self, snapshot_path: Option<&Path>) -> Result<(), Error> {
+        if matches!(snapshot_path, Some(p) if p.is_dir()) {
+            // TODO: Add Error in 2.0 as its breaking.
+            // Issue #1197
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                format!("Path is not a file: {:?}", snapshot_path.unwrap()),
```

Comment:
Do a proper match instead of `matches`

Reply:
Or put a comment for that unwrap, but unwrapping when we could match feels a bit dumb 🤷🏻‍♂️

Reply:
you even resolved the conversation where we changed it: https://github.com/iotaledger/iota-sdk/pull/1187#discussion_r1331513450

Reply:
Sorry my bad, I didn't see from the preview that there was an unwrap. Yes I would revert, I think it's worse than before.

Reply:
Or as I said, add a comment but it feels wrong to unwrap when there is a logic way to avoid it

Reply:
reverted